src/main/java/com/learning/pages/swaglabs/SwagLabsLoginPage.java

```diff
@@ -0,0 +1,63 @@
+package com.learning.pages.swaglabs;
+
+import com.learning.configuration.PropertiesReader;
+import com.learning.pages.PageBase;
+import java.util.Objects;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class SwagLabsLoginPage extends PageBase {
+
+  private final PropertiesReader propertiesReader;
+  private static final String SWAGLABS_URL_PROPERTIES_NAME = "swagLabsURL";
+
+  @FindBy(id = "user-name")
+  private WebElement usernameTextBox;
+  @FindBy(id = "password")
+  private WebElement passwordTextBox;
+  @FindBy(tagName = "h3")
+  private WebElement errorButton;
+  @FindBy(css = "input#login-button")
+  private WebElement loginButton;
+
+  public SwagLabsLoginPage(WebDriver driver, PropertiesReader propertiesReader) {
+    super(driver);
+    this.propertiesReader = propertiesReader;
+  }
+
+  public void openSwagLabsWebsite() {
+    driver.get(getSwagLabsURL());
+    waitForPageToLoad();
+  }
+
+  public void performLogin(String username, String password) {
+    usernameTextBox.sendKeys(username);
+    passwordTextBox.sendKeys(password);
+    loginButton.click();
+  }
+
+  public String getErrorText() {
+    return errorButton.getText();
+  }
+
+  @Override
+  public boolean waitForPageToLoad() {
+    WebDriverWait wait = new WebDriverWait(driver, 5);
+    return wait.until(ExpectedConditions
+        .visibilityOfAllElements(usernameTextBox, passwordTextBox,
+            loginButton))
+        .stream()
+        .allMatch(Objects::nonNull);
+  }
+
+  private String getSwagLabsURL() {
+    return propertiesReader.getProperty(SWAGLABS_URL_PROPERTIES_NAME);
+  }
+
+  public String getCurrentUrl() {
```

Comment:
I think this is one of those cases where you should perform operation on `Browser/Driver` object directly instead of PageObject class.
There should be as few cases as possible that would need browser/driver in tests, but I think this is one of them.

Reply:
Ok I get the idea that this method not belongs to PageObject class since it is not it's own URL.